### PR TITLE
add fork to child_process

### PR DESCRIPTION
**Pull request by sirenkovladd**
Copy from https://github.com/nodejs/node/blob/v20.5.0/lib/child_process.js#L101-L173
and corrected for `bun`

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
did you try it and see if it works?

**Comment by sirenkovladd** [comment]
Fixes https://github.com/oven-sh/bun/issues/3771 https://github.com/oven-sh/bun/issues/3620

**Comment by sirenkovladd** [comment]
> did you try it and see if it works?

Yes, for `dev` of course)
currently generating for release, may take some time

**Comment by Jarred-Sumner** [comment]
How does it communicate with the parent process and support sending data? The native code parts don't support additional file descriptors I don't think. Or is IPC less commonly used and this is more used as a way to run a file?

File: src/js/out/modules/node/child_process.js [hunk 1]
```diff
@@ -705,7 +745,6 @@ var child_process_default = {
 export {
   spawnSync,
   spawn,
-  fork,
```
Comment (birkskyum) [inline src/js/out/modules/node/child_process.js]: How come the named export is removed?

Comment (sirenkovladd) [reply src/js/out/modules/node/child_process.js]: I lost the export on the function, i fixed it

**Comment by Jarred-Sumner** [comment]
Can you include automated tests? These can be copied from Node. See https://github.com/oven-sh/bun/blob/main/test/js/node/child_process/child_process-node.test.js#L21  for an example

**Comment by sirenkovladd** [comment]
Ok, actually I found the blocks for this PR, it's the `process.send` method

Without it I can't send messages between processes

But I think this PR can be accepted and we can make a issue on solve `process.send` / `process.on('message')`

**Comment by sirenkovladd** [comment]
### What does this PR do?

Implement method `fork` in `child_process`

the fork method also needs methods for handling messages (process.send, process.on('message'), ChildProcess.send) to be fully compatible with node

- [X] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?



- [X] I wrote automated tests
- [X] I tested in my project which used `fork` from `node`

If JavaScript/TypeScript modules or builtins changed:

- [ ] I ran `make js` and committed the transpiled changes
(No rule to make target 'js')
- [X] I or my editor ran Prettier on the changed files (or I ran `bun fmt`)
- [X] I included a test for the new code, or an existing test covers it



If new methods, getters, or setters were added to a publicly exposed class:

- [ ] I added TypeScript types for the new methods, getters, or setters
- [X] The types already existed

If dependencies in tests changed:

- [X] I made sure that specific versions of dependencies are used instead of ranged or tagged versions

File: test/js/node/child_process/child_process-node.test.js [hunk 2]
```diff
@@ -483,3 +492,261 @@ describe("child_process double pipe", () => {
...
+  describe.todo("no-shell", () => {
+    // https://github.com/nodejs/node/blob/v20.5.0/test/parallel/test-child-process-fork-no-shell.js
+  });
+  describe.todo("ref", () => {
+    // https://github.com/nodejs/node/blob/v20.5.0/test/parallel/test-child-process-fork-ref.js
+  });
+  describe.todo("stdio", () => {
+    // https://github.com/nodejs/node/blob/v20.5.0/test/parallel/test-child-process-fork-stdio.js
+  });
+  describe.only("fork", () => {
```
Comment (Jarred-Sumner) [inline test/js/node/child_process/child_process-node.test.js]: It should not be using .only, instead .skip

**Comment by Jarred-Sumner** [comment]
This looks good, minus the usage of describe.only

**Review by Jarred-Sumner — APPROVED** [review]

File: test/js/node/child_process/child_process-node.test.js (hunk 2 above)
Comment (sirenkovladd) [reply test/js/node/child_process/child_process-node.test.js]: Fix it, thank you for finding this

**Comment by Jarred-Sumner** [comment]
The test for message should just be removed entirely since none of that works and a test which immediately calls done should be skipped since it does nothing and it won’t be entirely clear to future readers why the test exists or what it was supposed to do 

That being said, this is an improvement over the status quo so thank you and I’ll merge it now

**Pull request merged**